tests/pytest_testing_cosserat_curve.py

```diff
@@ -0,0 +1,257 @@
+
+    # Rotate the curve around its center point
+    pos_1, q_1 = curve.get_centerline_position_and_rotation(0.0)
+    curve.rotate(Rotation.from_quaternion(quaternion.as_float_array(q_1)), origin=pos_1)
+
+    # Get the points and rotations at certain points
+    t = list(map(float, range(-10, 30, 5)))
+    sol_half_pos, sol_half_q = curve.get_centerline_positions_and_rotations(
+        t, factor=0.5
+    )
+    sol_full_pos, sol_full_q = curve.get_centerline_positions_and_rotations(
+        t, factor=1.0
+    )
+
+    def load_compare(name):
+        """Load the compare files and return a numpy array"""
+        return np.loadtxt(
+            os.path.join(testing_input, f"{get_pytest_test_name()}_{name}.txt")
+        )
+
+    assert np.allclose(sol_half_pos, load_compare("pos_half_ref"), rtol=1e-14)
+    assert np.allclose(
+        quaternion.as_float_array(sol_half_q), load_compare("q_half_ref"), rtol=1e-14
+    )
+
+    assert np.allclose(sol_full_pos, load_compare("pos_full_ref"), rtol=1e-14)
+    assert np.allclose(
+        quaternion.as_float_array(sol_full_q), load_compare("q_full_ref"), rtol=1e-14
+    )
+
+
+def test_cosserat_curve_vtk_representation():
+    """Test the vtk representation of the Cosserat curve"""
+
+    result_name = os.path.join(testing_temp, get_pytest_test_name() + ".vtu")
+    curve = load_cosserat_curve_from_file()
+    pv.UnstructuredGrid(curve.get_pyvista_polyline()).save(result_name)
+    compare_vtk_pytest(
+        os.path.join(testing_input, get_pytest_test_name() + ".vtu"),
+        result_name,
+    )
+
+
+def test_cosserat_curve_project_point():
+    """Test that the project point function works as expected"""
+
+    # Load the curve
+    curve = load_cosserat_curve_from_file()
+
+    # Translate the curve so that the start is at the origin
+    curve.translate(-curve.centerline_interpolation(0.0))
+
+    # Check the projection results
+    tol = 1e-14
+    t_ref = 4.264045157204052
+    assert t_ref == pytest.approx(curve.project_point([-5, 1, 1]), rel=tol)
+    assert t_ref == pytest.approx(curve.project_point([-5, 1, 1], t0=2.0), rel=tol)
+    assert t_ref == pytest.approx(curve.project_point([-5, 1, 1], t0=4.0), rel=tol)
+
+
+def test_cosserat_mesh_transformation():
+    """Test that the get_mesh_transformation function works as expected"""
+
+    curve = load_cosserat_curve_from_file()
+    pos, rot = curve.get_centerline_position_and_rotation(0)
+    rot = Rotation.from_quaternion(quaternion.as_float_array(rot))
+    curve.translate(-pos)
+    curve.translate([1, 2, 3])
+
+    mesh = create_beam_solid_input_file()
+    pos, rot = get_mesh_transformation(
+        curve,
+        mesh.nodes,
+        origin=[0.5, 1.0, -1.0],
+        reference_rotation=(
+            Rotation([0, 0, 1], -0.5 * np.pi) * Rotation([0, 1, 0], -0.5 * np.pi)
+        ),
+        n_steps=3,
+    )
+
+    # Save as json:
+    # with open("name.json", "w") as f:
+    #     json.dump(np_array.tolist(), f, indent=2)
```

Comment:
I would leave this one, as here, `np_array` is a 3D array, which is not straightforward to write to disk. So in case the reference results need to be regenerated, I would leave the commented code here.